boards/arm/gd32f350r_eval/doc/index.rst

```diff
@@ -0,0 +1,134 @@
+.. _gd32f350r_eval:
+
+GigaDevice GD32F350R-EVAL
+#########################
+
+Overview
+********
+
+The GD32F350R-EVAL board is a hardware platform that enables design and debug
+of the GigaDevice F350 Cortex-M4F High Performance MCU.
+
+The GD32F350RBT6 features a single-core ARM Cortex-M4F MCU which can run up
+to 108-MHz with flash accesses zero wait states, 128kB of Flash, 16kB of
+SRAM and 55 GPIOs.
+
+.. image:: img/gd32f350r_eval.png
+     :align: center
+     :alt: gd32f350r_eval
+
+Hardware
+********
+
+- GD32F350RBT6 MCU
+- AT24C02C 2Kb EEPROM
+- 4 x User LEDs
+- 4 x User Push buttons
+- 1 x USART (RS-232 at J2 connector)
+- 1 x POT connected to an ADC input
+- Headphone interface
+- Micro SD Card Interface
+- 2.4'' TFT-LCD (36x48)
+- GD-Link on board programmer
+- J-Link/SWD connector
+
+For more information about the GD32F350 SoC and GD32F350R-EVAL board:
+
+- `GigaDevice Cortex-M4F Stretch Performance SoC Website`_
+- `GD32F350xx Datasheet`_
+- `GD32F3x0 User Manual`_
+- `GD32F350R-EVAL User Manual`_
+
+Supported Features
+==================
+
+The board configuration supports the following hardware features:
+
++-----------+------------+---------------------------------------+
+| Interface | Controller | Driver/Component                      |
++===========+============+=======================================+
+| NVIC      | on-chip    | nested vectored interrupt controller  |
++-----------+------------+---------------------------------------+
+| SYSTICK   | on-chip    | system clock                          |
++-----------+------------+---------------------------------------+
+| USART     | on-chip    | serial port-polling                   |
+|           |            | serial port-interrupt                 |
++-----------+------------+---------------------------------------+
+| PINMUX    | on-chip    | pinmux                                |
++-----------+------------+---------------------------------------+
```

Comment:
suggestion: create a table like https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/gd32f450i_eval/doc/index.rst#supported-features (adds links to relevant Kconfig/compatibles, useful for developers)

Reply:
done.